upi/vsphere/powercli/upi-functions.ps1

```diff
@@ -47,28 +48,28 @@ function New-OpenShiftVM {
     # If storage policy is set, lets pull the mo ref
     if ($NULL -ne $StoragePolicy -and $StoragePolicy -ne "")
     {
-        $storagePolicyRef = Get-SpbmStoragePolicy -Id $StoragePolicy
+        $storagePolicyRef = Get-SpbmStoragePolicy -Server $Server -Id $StoragePolicy
         $args["StoragePolicy"] = $storagePolicyRef
     }
 
     # Clone the virtual machine from the imported template
     # $vm = New-VM -VM $Template -Name $Name -Datastore $Datastore -ResourcePool $ResourcePool #-Location $Folder #-LinkedClone -ReferenceSnapshot $Snapshot
-    $vm = New-VM -VM $Template @args
+    $vm = New-VM -Server $Server -VM $Template @args
 
     # Assign tag so we can later clean up
-    New-TagAssignment -Entity $vm -Tag $Tag > $null
+    New-TagAssignment -Server $Server -Entity $vm -Tag $Tag > $null
 
     # Update VM specs.  New-VM does not honor the passed in parameters due to Template being used.
     if ($null -ne $MemoryMB -And $null -ne $NumCpu)
     {
-        Set-VM -VM $vm -MemoryMB $MemoryMB -NumCpu $NumCpu -CoresPerSocket 4 -Confirm:$false > $null
+        Set-VM -Server $Server -VM $vm -MemoryMB $MemoryMB -NumCpu $NumCpu -CoresPerSocket 4 -Confirm:$false > $null
     }
     #Get-HardDisk -VM $vm | Select-Object -First 1 | Set-HardDisk -CapacityGB 120 -Confirm:$false > $null
     updateDisk -VM $vm -CapacityGB 120
 
     # Configure Network (Assuming template networking may not be correct if shared across clusters)
-    $pg = Get-VirtualPortgroup -Name $Network -VMHost $(Get-VMHost -VM $vm) 2> $null
-    $vm | Get-NetworkAdapter | Set-NetworkAdapter -Portgroup $pg -confirm:$false > $null
+    $pg = Get-VirtualPortgroup -Server $Server -Name $Network -VMHost $(Get-VMHost -VM $vm) 2> $null
+    $vm | Get-NetworkAdapter -Server $Server | Set-NetworkAdapter -Server $Server -Portgroup $pg -confirm:$false > $null
 
     # Assign advanced settings
     New-AdvancedSetting -Entity $vm -name "disk.enableUUID" -value "TRUE" -confirm:$false -Force > $null
```

Comment:
do only certain cmdlets need `-server`?

Reply:
All of the vSphere cmdlets have it.  I know in some cases, it can pull out the server from the object passed in, but I was being super cautious just in case.